pkg/gardenlet/controller/managedseed/managedseed_reconciler.go

```diff
@@ -0,0 +1,280 @@
+		managedSeedLogger.Errorf("Could not reconcile ManagedSeed deletion: %+v", err)
+
+		// Record an event
+		lastError := &gardencorev1beta1.LastError{
+			Codes:       gardencorev1beta1helper.ExtractErrorCodes(err),
+			Description: "Could not reconcile deletion: " + err.Error(),
+		}
+		r.recorder.Eventf(managedSeed, corev1.EventTypeWarning, gardencorev1beta1.EventDeleteError, "%s", lastError.Description)
+
+		// Update status to Error
+		if updateErr := updateStatusError(r.ctx, gardenClient.DirectClient(), managedSeed, lastError.Description, lastError); updateErr != nil {
+			managedSeedLogger.Errorf("Could not update ManagedSeed status after deletion error: %+v", updateErr)
+			return reconcile.Result{}, updateErr
+		}
+
+		// Retrun error result
+		return reconcile.Result{}, errors.New(lastError.Description)
+	}
+	managedSeedLogger.Debug("Successfully reconciled ManagedSeed deletion")
+
+	// Update status to Succeeded
+	if updateErr := updateStatusSucceeded(r.ctx, gardenClient.DirectClient(), managedSeed, "Deleted successfully"); updateErr != nil {
+		managedSeedLogger.Errorf("Could not update ManagedSeed status after deletion success: %+v", updateErr)
+		return reconcile.Result{}, updateErr
+	}
+
+	// Return success result and remove finalizer
+	return reconcile.Result{}, controllerutils.RemoveGardenerFinalizer(r.ctx, gardenClient.DirectClient(), managedSeed)
+}
+
+func updateStatusProcessing(ctx context.Context, c client.Client, managedSeed *seedmanagementv1alpha1.ManagedSeed, description string) error {
+	return updateStatus(ctx, c, managedSeed, description, gardencorev1beta1.LastOperationStateProcessing, 0, false, false, nil)
+}
+
+func updateStatusError(ctx context.Context, c client.Client, managedSeed *seedmanagementv1alpha1.ManagedSeed, description string, lastError *gardencorev1beta1.LastError) error {
+	return updateStatus(ctx, c, managedSeed, description, gardencorev1beta1.LastOperationStateError, 0, false, true, lastError)
+}
+
+func updateStatusSucceeded(ctx context.Context, c client.Client, managedSeed *seedmanagementv1alpha1.ManagedSeed, description string) error {
+	return updateStatus(ctx, c, managedSeed, description, gardencorev1beta1.LastOperationStateSucceeded, 100, true, true, nil)
+}
+
+func updateStatusPending(ctx context.Context, c client.Client, managedSeed *seedmanagementv1alpha1.ManagedSeed, description string) error {
+	return updateStatus(ctx, c, managedSeed, description, gardencorev1beta1.LastOperationStatePending, 0, true, false, nil)
+}
+
+func updateStatus(ctx context.Context, c client.Client, managedSeed *seedmanagementv1alpha1.ManagedSeed, description string, state gardencorev1beta1.LastOperationState, progress int32, updateObservedGeneration, updateLastError bool, lastError *gardencorev1beta1.LastError) error {
+	return kutil.TryUpdateStatus(ctx, retry.DefaultBackoff, c, managedSeed, func() error {
+		managedSeed.Status.LastOperation = &gardencorev1beta1.LastOperation{
```

Comment:
As proposed earlier, let's switch to (two) conditions for the managed seed status, e.g. `ShootReconciled` and `GardenletDeployed`/`SeedRegistered` or something like that.

Reply:
OK, I will add the conditions, how about `LastError` and `LastOperation`, should I remove them?